### PR TITLE
Void cache.

### PR DESCRIPTION
**Pull request by makasim**
I found this useful in integration tests.  When you perform a test case inside a transaction and rollback any changes at the end but result cache still contains modified data.

I'll add tests if you find this useful.

### CODE REVIEW

File: lib/Doctrine/Common/Cache/VoidCache.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getStats()
+    {
+        return;
+    }
+}
```
Comment (GrahamCampbell) [inline lib/Doctrine/Common/Cache/VoidCache.php]: files must end in a single new line

**Comment by Ocramius** [comment]
I'd actually create a `VoidCache` based on the `CacheProvider` if possible. It can't use the base test for it, though.

**Comment by Ocramius** [comment]
Note that this is a good feature, but you really need tests for it to be accepted ;-)

**Comment by makasim** [comment]
> you really need tests for it to be accepted ;-)

@Ocramius np

**Comment by makasim** [comment]
@Ocramius done, Could you please review it?

File: tests/Doctrine/Tests/Common/Cache/VoidCacheTest.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Doctrine\Tests\Common\Cache;
+
+use Doctrine\Common\Cache\VoidCache;
+
+class VoidCacheTest extends \PHPUnit_Framework_TestCase
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/VoidCacheTest.php]: Please add coverage annotations

File: lib/Doctrine/Common/Cache/VoidCache.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function doSave($id, $data, $lifeTime = 0)
+    {
+        return true;
```
Comment (Ocramius) [inline lib/Doctrine/Common/Cache/VoidCache.php]: Is it actually a success?

Comment (Ocramius) [inline lib/Doctrine/Common/Cache/VoidCache.php]: Hmmm, yeah, it probably is (non-exceptional situation)

File: tests/Doctrine/Tests/Common/Cache/VoidCacheTest.php (hunk 2 above)
Comment (makasim) [inline tests/Doctrine/Tests/Common/Cache/VoidCacheTest.php]: hm, cannot find any examples in other files... what is this for?

Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Cache/VoidCacheTest.php]: @makasim `@covers \Doctrine\Common\Cache\VoidCache`

File: lib/Doctrine/Common/Cache/VoidCache.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Doctrine\Common\Cache;
```
Comment (deeky666) [inline lib/Doctrine/Common/Cache/VoidCache.php]: Doctrine File DocBlock and newline before

File: tests/Doctrine/Tests/Common/Cache/VoidCacheTest.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Doctrine\Tests\Common\Cache;
+
+use Doctrine\Common\Cache\VoidCache;
+
+/**
+ * @covers \Doctrine\Common\Cache\VoidCache
+ */
```
Comment (makasim) [inline tests/Doctrine/Tests/Common/Cache/VoidCacheTest.php]: @Ocramius like this?

File: lib/Doctrine/Common/Cache/VoidCache.php [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+class VoidCache extends CacheProvider
```
Comment (deeky666) [inline lib/Doctrine/Common/Cache/VoidCache.php]: A class-level DocBlock with a description of that class would be good.

File: lib/Doctrine/Common/Cache/VoidCache.php (hunk 4 above)
Comment (makasim) [inline lib/Doctrine/Common/Cache/VoidCache.php]: @deeky666 done

**Comment by deeky666** [comment]
Nice feature!

**Comment by beberlei** [comment]
The class block should contain a description what to use this cache for.

And not asking for a change yet, but wouldn't `NullCache` or `DevNullCache` be better names?

**Comment by guilhermeblanco** [comment]
@beberlei I'd add to the list of options: `BlackholeCache`

**Comment by makasim** [comment]
@beberlei @guilhermeblanco Let me know the class you prefer most and I rename it

**Comment by makasim** [comment]
> The class block should contain a description what to use this cache for.

done 

@Ocramius @beberlei @guilhermeblanco Is it good now?

**Comment by Ocramius** [comment]
@makasim merging, thanks!

**Pull request merged**